bigcode_eval/evaluator.py

```diff
@@ -66,7 +66,7 @@ def generate_text(self, task_name, intermediate_generations=None):
         intermediate_save_generations_path = f"{os.path.splitext(self.args.save_generations_path)[0]}_{task_name}_intermediate.json"
         curr_sample_idx = len(curr_generations)
 
-        generations = parallel_generations(
+        generation_dict = parallel_generations(
```

Comment:
Can we avoid changing the return type here to minimize fundamental changes to the existing API?

For instance, right now this function already accepts params like `save_every_k_tasks` and `intermediate_save_generations_path`, maybe we can add a new optional param like `generation_with_fewshot_output_dir`